File: src/dataframe/rust/stats/distributions/geometric.test.R
#!/usr/bin/env Rscript

# Geometric distribution test script
# Usage: Rscript geometric.test.R <function> <x_or_p> <prob> [lower_tail] [log_p]

source("test-helpers.R")

# Define geometric distribution functions
geometric_density <- function(x, prob, log = FALSE) {
  dgeom(x, prob, log = log)
}

geometric_cumulative <- function(x, prob, lower.tail = TRUE, log.p = FALSE) {
  pgeom(x, prob, lower.tail = lower.tail, log.p = log.p)
}

geometric_quantile <- function(p, prob, lower.tail = TRUE, log.p = FALSE) {
  qgeom(p, prob, lower.tail = lower.tail, log.p = log.p)
}

geometric_random <- function(n, prob) {
  rgeom(n, prob)
}

# Run the test
run_distribution_test(geometric_density, geometric_cumulative, geometric_quantile, geometric_random)

File: src/dataframe/rust/stats/distributions/geometric.test.ts
#!/usr/bin/env -S deno run --allow-read --allow-run

import { callR, printResults, runComparison } from "./test-helpers.ts";

async function main() {
  console.log("Testing R vs Rust geometric functions...\n");

  const testCases = [
    { func: "dgeom", distribution: "geometric", args: [2, 0.3, 0] },
    { func: "pgeom", distribution: "geometric", args: [2, 0.3, 1, 0] },
    { func: "qgeom", distribution: "geometric", args: [0.657, 0.3, 1, 0] },
  ];

  const results = await runComparison(testCases);
  printResults(results);

  // Test rgeom separately since it's not available in Rust interface
  console.log("\nR-only tests:");
  try {
    const rgeomResult = await callR("rgeom", "geometric", 1, 0.3);
    console.log(`rgeom(0.3) = ${rgeomResult}`);
  } catch (error) {
    console.log(`rgeom: Error - ${(error as Error).message}`);
  }
}

if (import.meta.main) {
  main().catch(console.error);
}
